classes/migration/upgrade/v3_4_0/I7264_UpdateEmailTemplates.inc.php

```diff
@@ -0,0 +1,414 @@
+        ], function (Builder $q) {
+            $q->selectRaw('? as email_key', ['USER_VALIDATE_SITE'])
+                ->addSelect('locale', 'subject', 'body', 'description')
+                ->from('email_templates_default_data')
+                ->where('email_key', '=', 'USER_VALIDATE_CONTEXT');
+        });
+
+        // Replace all template variables
+        $oldNewVariablesMap = $this->oldNewVariablesMap();
+        $this->renameTemplateVariables($oldNewVariablesMap);
+    }
+
+    /**
+     * Reverse the downgrades
+     */
+    public function down(): void
+    {
+        // Revert variables renaming
+        $newOldVariablesMap = [];
+        foreach ($this->oldNewVariablesMap() as $emailKey => $variablesMap) {
+            $newOldVariablesMap[$emailKey] = array_flip($variablesMap);
+        }
+        $this->renameTemplateVariables($newOldVariablesMap);
+
+        // Revert renaming email template keys
+        foreach ([
+             DB::table('email_templates'),
+             DB::table('email_templates_default'),
+             DB::table('email_templates_default_data')
+         ] as $tableContainingKeys) {
+            $tableContainingKeys->where('email_key', 'USER_VALIDATE_CONTEXT')
+                ->update(['email_key' => 'USER_VALIDATE']);
+
+            $tableContainingKeys->where('email_key', 'ISSUE_PUBLISH_NOTIFY')
+                ->update(['email_key' => 'PUBLISH_NOTIFY']);
+
+            $tableContainingKeys->where('email_key', 'REVIEW_RESPONSE_OVERDUE_AUTO')
+                ->update(['email_key' => 'REVIEW_REQUEST_REMIND_AUTO']);
+
+            $tableContainingKeys->where('email_key', 'REVIEW_RESPONSE_OVERDUE_AUTO_ONECLICK')
+                ->update(['email_key' => 'REVIEW_REQUEST_REMIND_AUTO_ONECLICK']);
+
+            $tableContainingKeys->where('email_key', 'USER_VALIDATE_SITE')->delete();
+        }
+
+    }
+
+    /**
+     * Replaces email template variables in templates' subject and body
+     */
+    protected function renameTemplateVariables(array $oldNewVariablesMap): void
+    {
+        foreach ($oldNewVariablesMap as $emailKey => $variablesMap) {
+            $data = DB::table('email_templates_default_data')->where('email_key', $emailKey)->get();
+            $variables = [];
+            $replacements = [];
+            foreach ($variablesMap as $key => $value) {
+                $variables[] = '/\{\$' . $key . '\}/';
+                $replacements[] = '{$' . $value . '}';
+            }
+
+            $data->each(function (stdClass $entry) use ($variables, $replacements) {
+                $subject = preg_replace($variables, $replacements, $entry->subject);
+                $body = preg_replace($variables, $replacements, $entry->body);
+                DB::table('email_templates_default_data')
+                    ->where('email_key', $entry->{'email_key'})
+                    ->where('locale', $entry->{'locale'})
+                    ->update(['subject' => $subject, 'body' => $body]);
```

Comment:
I _think_ you can do just `body`. I don't think we have supported variables in subjects up to now.